File: LeetCodeProblemsJava/src/Problems/LeetCode3751TotalWavinessOfNumbersInRangeI.java
package Problems;

public class LeetCode3751TotalWavinessOfNumbersInRangeI {
    public int totalWaviness(int num1, int num2) {
        int result = 0;
        for (int i = num1; i <= num2; i++) {
            result += getWaviness(i);
        }
        return result;
    }
    private int getWaviness(int num) {
        int previous=num%10;
        num/=10;
        int current=0;
        int result=0;
        int next=0;
        while (num>0){
            current=num%10;
            num/=10;
            if(num==0)
                break;
            next=num%10;
            if( current>previous && current>next)
                result++;
            if( current<previous && current<next)
                result++;
            previous=current;
        }
        return result;
    }
}

File: LeetCodeProblemsGo/LeetCode3751TotalWavinessOfNumbersInRangeI/LeetCode3751TotalWavinessOfNumbersInRangeI.go
package LeetCode3751TotalWavinessOfNumbersInRangeI

func totalWaviness(num1 int, num2 int) int {
	result := 0
	for ; num1 <= num2; num1++ {
		result += getWaviness(num1)
	}
	return result
}

func getWaviness(num int) int {
	previous := num % 10
	num /= 10
	next, result, current := 0, 0, 0
	for num > 0 {
		current = num % 10
		num /= 10
		if num == 0 {
			return result
		}
		next = num % 10
		if (current > previous && current > next) || (current < previous && current < next) {
			result++
		}
		previous = current
	}
	return result
}
